fix(parser): reject `using` / `await using` in a switch `case` / `default` clause

sapphi-red opened the pull request:
`using` / `await using` directly nested within a `case` or `default` clause are now disallowed (https://github.com/rbuckton/ecma262/pull/14). This PR implements this change.

Note that TypeScript does not disallow them yet (https://github.com/microsoft/TypeScript/issues/62708) and the test failing in `tasks/coverage/snapshots/parser_typescript.snap` is caused by that.

Boshen commented:
Saw the title: https://github.com/tc39/test262/issues/4498

Boshen commented:
Let's merge to avoid merge conflicts. Also because babel and acorn has implemented this.

sapphi-red committed:
fix(parser): reject `using` / `await using` in a switch `case` / `default` clause (#15225)

`using` / `await using` directly nested within a `case` or `default` clause are now disallowed (https://github.com/rbuckton/ecma262/pull/14). This PR implements this change.

Note that TypeScript does not disallow them yet (https://github.com/microsoft/TypeScript/issues/62708) and the test failing in `tasks/coverage/snapshots/parser_typescript.snap` is caused by that.
